### PR TITLE
Tests: Increase coverage of the line followers

### PR DESCRIPTION
**Pull request by lhelwerd**
Also change a few interface parts, namely making the USB manager
actually optional in the non-abstract classes and raising TypeErrors
instead of ValueErrors.

### CODE REVIEW

File: tests/location_line_follower_arduino.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    def test_initialization(self):
+        # The settings argument must be `Arguments` or `Settings`
+        with self.assertRaises(TypeError):
+            Line_Follower_Arduino(
+                self.location, self.direction, self.mock_callback,
+                None, self.thread_manager, usb_manager=self.usb_manager, delay=0
+            )
+
+        # The settings argument must be `Arguments` or `Settings`
```
Comment (timvandermeij) [inline tests/location_line_follower_arduino.py]: Incorrect comment

File: tests/location_line_follower_arduino.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            Line_Follower_Arduino(
+                self.location, self.direction, self.mock_callback,
+                self.settings, self.thread_manager, usb_manager=None, delay=0
+            )
+
+        Line_Follower_Arduino(
+            self.location, self.direction, self.mock_callback, self.arguments,
+            self.thread_manager, usb_manager=self.usb_manager, delay=0
+        )
+
```
Comment (timvandermeij) [inline tests/location_line_follower_arduino.py:52]: Nit: unnecessary newline

File: tests/location_line_follower_arduino.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            )
+
+        # The settings argument must be `Arguments` or `Settings`
+        with self.assertRaisesRegexp(TypeError, "'usb_manager' must be provided"):
+            Line_Follower_Arduino(
+                self.location, self.direction, self.mock_callback,
+                self.settings, self.thread_manager, usb_manager=None, delay=0
+            )
+
+        Line_Follower_Arduino(
```
Comment (timvandermeij) [inline tests/location_line_follower_arduino.py:48]: This is already tested in `setUp`?

File: tests/location_line_follower_arduino.py (hunk 2 above)
Comment (lhelwerd) [inline tests/location_line_follower_arduino.py:52]: Removed the double newline.

**Pull request merged**